feat: rename sandbox to local network

alejoamiras opened the pull request:
- Renames the `--sandbox` flag to `--local-network`
- Updates docs to reflect that

<img width="833" height="248" alt="image" src="https://github.com/user-attachments/assets/4a6c9c41-1074-4138-9ff6-9aeef60a673a" />

spalladino reviewed (commented):
LGTM except for the remaining sandbox mention on `aztec-up/bin/aztec`. I also didn't look into docs or any md files.

critesjosh reviewed (approved):
looks good to me on the docs side. maybe we just want to update this line?

https://github.com/AztecProtocol/aztec-packages/blob/ac94ba0368ae0082f6ceceba9ed966eb5763fdb1/aztec-up/bin/aztec#L325-L326

alejoamiras commented:
> [/aztec-up/bin/aztec](https://github.com/AztecProtocol/aztec-packages/blob/ac94ba0368ae0082f6ceceba9ed966eb5763fdb1/aztec-up/bin/aztec#L325-L326)

I've left a message on engineering channel referencing this, but I think it's a bit more meta and don't really want to make a mistake. This is clearly dead code, because `aztec-sandbox` hasn't existed for a while.

alejoamiras committed:
feat: rename sandbox to local network

- Renames the `--sandbox` flag to `--local-network`
- Updates docs to reflect that

<img width="833" height="248" alt="image" src="https://github.com/user-attachments/assets/4a6c9c41-1074-4138-9ff6-9aeef60a673a" />

Co-authored-by: Alejo Amiras <alejo.amiras@gmail.com>
